src/lib/SharedLogic.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.27;
+
+import { ConstructorLogic } from "./ConstructorLogic.sol";
+import { IdLib } from "./IdLib.sol";
+
+import { SafeTransferLib } from "solady/utils/SafeTransferLib.sol";
+
+/**
+ * @title SharedLogic
+ * @notice Inherited contract implementing logic for internal functions with
+ * low-level shared logic for processing transfers and withdrawals.
+ */
+contract SharedLogic is ConstructorLogic {
+    using IdLib for uint256;
+    using SafeTransferLib for address;
+
+    // Storage slot seed for ERC6909 state, used in computing balance slots.
+    uint256 private constant _ERC6909_MASTER_SLOT_SEED = 0xedcaa89a82293940;
```

Comment:
what is this value? where did it come from?

Reply:
this is from solady's ERC6909 implementation, since the `_release` and `_withdraw` functions use the same general pathways as the standard transfers (but skip allowance checks and transfer hooks)